### PR TITLE
Add bulk_update_with_history 

### PR DESCRIPTION
**Pull request by jihoon796**
I have yet to add tests or documentation. Once a maintainer gives the green light, I'll finish writing all of the documentation, although I will need help writing tests. 

## Description
Added `bulk_update_with_history`, using Django 2.2's `bulk_update` feature. 

## Related Issue
https://github.com/treyhunner/django-simple-history/issues/528

## Motivation and Context
https://github.com/treyhunner/django-simple-history/issues/528

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run the `make format` command to format my code
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] I have added my name and/or github handle to `AUTHORS.rst`
- [x] I have added my change to `CHANGES.rst`
- [x] All new and existing tests passed.

### CODE REVIEW

File: simple_history/manager.py [hunk 1]
```diff
@@ -98,16 +98,20 @@ def _as_of_set(self, date):
...
 
+        history_type = "+"
+        if update:
+            history_type = '~'
+
         historical_instances = []
         for instance in objs:
             row = self.model(
-                history_date=getattr(instance, "_history_date", timezone.now()),
+                history_date=getattr(instance, "_history_date", now()),
```
Comment (rossmechanic) [inline simple_history/manager.py:111]: can you keep `timezone.now()`?

File: simple_history/manager.py [hunk 2]
```diff
@@ -98,16 +98,20 @@ def _as_of_set(self, date):
                 continue
             yield last_change.instance
 
-    def bulk_history_create(self, objs, batch_size=None):
+    def bulk_history_create(self, objs, batch_size=None, update=False):
```
Comment (rossmechanic) [inline simple_history/manager.py:101]: Add to the docstring please

**Comment by rossmechanic** [comment]
Also, since the codebase still technically supports Django 1.11, 2.0, and 2.1, just have the tests skip if if the django version is any of those. I'm going to remove support for them before the next release.

File: simple_history/manager.py (hunk 1 above)
Comment (jihoon796) [reply simple_history/manager.py:111]: Most definitely - will make the change.

File: simple_history/manager.py (hunk 2 above)
Comment (jihoon796) [reply simple_history/manager.py:101]: Done.

**Comment by jihoon796** [comment]
@rossmechanic ready for review. 

I use `django-simple-history` frequently, so happy to contribute.

File: docs/common_issues.rst [hunk 3]
```diff
@@ -42,8 +42,28 @@ can add `changeReason` on each instance:
     >>> Poll.history.get(id=data[0].id).history_change_reason
     'reason'
 
+Bulk Updating a Model with History (New)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (rossmechanic) [inline docs/common_issues.rst:46]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: docs/common_issues.rst [hunk 4]
```diff
@@ -5,7 +5,7 @@ Bulk Creating and Queryset Updating
 -----------------------------------
 ``django-simple-history`` functions by saving history using a ``post_save`` signal
 every time that an object with history is saved. However, for certain bulk
-operations, such as bulk_create_ and `queryset updates <https://docs.djangoproject.com/en/2.0/ref/models/querysets/#update>`_,
+operations, such as bulk_create_, `bulk_update, and `queryset updates <https://docs.djangoproject.com/en/2.2/ref/models/querysets/#update>`_,
```
Comment (rossmechanic) [inline docs/common_issues.rst:8]: the the link `bulk_update` should be:  https://docs.djangoproject.com/en/3.0/ref/models/querysets/#bulk-update

You can use the underscore notation like on bulk_create

File: docs/common_issues.rst [hunk 5]
```diff
@@ -42,8 +42,28 @@ can add `changeReason` on each instance:
...
+    >>>
+    >>> data = [Poll(id=x, question='Question ' + str(x), pub_date=now()) for x in range(1000)]
+    >>> objs = bulk_create_with_history(data, Poll, batch_size=500)
+    >>> for obj in objs: obj.question = 'Duplicate Questions'
+    >>> bulk_update_with_history(objs, Poll, ['question'], batch_size=500)        
+    >>> Poll.objects.first().question
+    'Duplicate Question'
 
-QuerySet Updates with History
+QuerySet Updates with History (Updated in Django 2.2)
```
Comment (rossmechanic) [inline docs/common_issues.rst:66]: extend the underline under this all the way. if you run `make docs` locally, you'll see the warning that come up if you don't do this

**Comment by rossmechanic** [comment]
@jihoon796 also, I just looked at the travis build https://travis-ci.org/github/treyhunner/django-simple-history. Looks like you need to run `make format` as well. Not sure why travis isn't linking with github checks 🤷

**Comment by jihoon796** [comment]
@rossmechanic updated the docs as requested and ran `make docs` and `make format` (this time I double-checked).

**Comment by rossmechanic** [comment]
https://travis-ci.org/github/treyhunner/django-simple-history/jobs/679104605
@jihoon796 looks like there's a trailing whitespace error

**Comment by jihoon796** [comment]
@rossmechanic got it - fixed the trailing whitespace error, so it should pass all checks now. 

When is the next planned release? Would love to use it as soon as possible.

**Comment by rossmechanic** [comment]
Cool. Looks like it's passing everything. 

I wasn't planning to do a release for another month (and get some changes in before doing a 3.0 release), which is why I said don't worry about backward compatibility. However, if you want to update this PR so that it's still compatible with django versions 1.11, 2.0 and 2.1, then I'd be happy to get this PR and #636 merged and released this weekend. The other options is to pip install this merge commit directly. What do you want to do?

**Comment by rossmechanic** [comment]
for reference, to make this backward compatible you just need to do the following within your `bulk_update_with_history` function:
```
if `django.VERSION < (2, 2, )`:
     raise NotImplementedError("bulk_update_with_history is only available on Django versions 2.2 and later")
```

**Comment by jihoon796** [comment]
@rossmechanic Would be great to get the release out as early as possible. I've updated this PR (added your snippet) so that it's still compatible with Django versions 1.11, 2.0 and 2.1. Ran the same tests as earlier as well and everything looks good. 

Also - thanks for the quick feedback!

**Pull request merged**